backend/core/src/application-flagged-sets/application-flagged-sets.service.ts

```diff
@@ -115,6 +124,69 @@ export class ApplicationFlaggedSetsService {
     }
   }
 
+  private async _getAfsesContainingApplicationId(
+    queryRunnery: QueryRunner,
+    applicationId: string
+  ): Promise<Array<{ id: string }>> {
+    const metadataArgsStorage = getMetadataArgsStorage().findJoinTable(
+      ApplicationFlaggedSet,
+      "applications"
+    )
+    const applicationsJunctionTableName = metadataArgsStorage.name
+    const afsMetadata = queryRunnery.connection.getMetadata(ApplicationFlaggedSet)
+    const applicationsMetadata = queryRunnery.connection.getMetadata(Application)
+    const query = `
+      SELECT DISTINCT afs.id FROM ${afsMetadata.tableName} afs 
+      INNER JOIN ( 
+          SELECT DISTINCT application_flagged_set_id, applications_id FROM ${applicationsJunctionTableName} 
+          WHERE ${applicationsJunctionTableName}.applications_id = $1
+      ) matching_afs ON afs.id = matching_afs.application_flagged_set_id
+      LEFT JOIN ${applicationsJunctionTableName} ON matching_afs.application_flagged_set_id = ${applicationsJunctionTableName}.application_flagged_set_id
+      LEFT JOIN ${applicationsMetadata.tableName} applications ON applications.id = ${applicationsJunctionTableName}.applications_id
```

Comment:
@pbn4 ,
I'm probably missing somethings but does the query here return the same as querying `applicationsJunctionTableName` directly?

If I take what's here and compare to the below query, they appear to return the same result.

```
select distinct application_flagged_set_id from ${applicationsJunctionTableName}
where applications_id = $1
```

Reply:
Good point, simplified. Thanks